### PR TITLE
Define database schema in models.py

### PR DESCRIPTION
**Pull request by dhimmel**
```
Co-authored-by: Dongbo Hu <dongbo.hu@gmail.com>
```

### CODE REVIEW

**Comment by dhimmel** [comment]
# Prior writey

img_20181101_112428

File: dj_hetmech/models.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    class Meta:
+        unique_together = ('metanode', 'identifier')
+
+
+class Metapath(models.Model):
+    abbreviation = models.CharField(primary_key=True, max_length=20)
+    verbose = models.CharField(max_length=200)
+    verbose_pretty = models.CharField(max_length=200)
+    source = models.ForeignKey(to='Metanode', on_delete=models.PROTECT, related_name='metapath_source')
+    target = models.ForeignKey(to='Metanode', on_delete=models.PROTECT, related_name='metapath_target')
```
Comment (dhimmel) [inline dj_hetmech/models.py:47]: Without `related_name`, I was getting the following errors:

```
dj_hetmech.Metapath.source: (fields.E304) Reverse accessor for 'Metapath.source' clashes with reverse accessor for 'Metapath.target'.
	HINT: Add or change a related_name argument to the definition for 'Metapath.source' or 'Metapath.target'.
dj_hetmech.Metapath.target: (fields.E304) Reverse accessor for 'Metapath.target' clashes with reverse accessor for 'Metapath.source'.
	HINT: Add or change a related_name argument to the definition for 'Metapath.target' or 'Metapath.source'.
```

Seems like we could also set `related_name='+'` to suppress backwards relations.
https://docs.djangoproject.com/en/2.1/ref/models/fields/#django.db.models.ForeignKey.related_name

However, I don't fully understand backwards relations and when we would use them.

**Comment by dhimmel** [comment]
Beautiful right?

**Review by dongbohu — APPROVED** [review]
Looks great!

**Comment by dongbohu** [comment]
@dhimmel Please feel free to close this PR. Thanks.

**Comment by dhimmel** [comment]
> Please feel free to close this PR

I think you mean merge (:

**Pull request merged**